source/fe/fe_pyramid_p.cc

```diff
@@ -103,6 +103,47 @@ FE_PyramidPoly<dim, spacedim>::FE_PyramidPoly(
       this->unit_support_points.emplace_back(-1.0, +1.0, 0.0);
       this->unit_support_points.emplace_back(+1.0, +1.0, 0.0);
       this->unit_support_points.emplace_back(+0.0, +0.0, 1.0);
+
+      this->unit_face_support_points.resize(5);
+
+      for (unsigned int f = 0; f < 5; ++f)
+        {
+          if (f == 0)
+            {
+              this->unit_face_support_points[f].emplace_back(0.0, 0.0);
+              this->unit_face_support_points[f].emplace_back(1.0, 0.0);
+              this->unit_face_support_points[f].emplace_back(0.0, 1.0);
+              this->unit_face_support_points[f].emplace_back(1.0, 1.0);
+            }
+          else
+            {
+              this->unit_face_support_points[f].emplace_back(0.0, 0.0);
+              this->unit_face_support_points[f].emplace_back(1.0, 0.0);
+              this->unit_face_support_points[f].emplace_back(0.0, 1.0);
+            }
+        }
+    }
+}
+
+
+
+template <int dim, int spacedim>
+void
+FE_PyramidPoly<dim, spacedim>::
+  convert_generalized_support_point_values_to_dof_values(
+    const std::vector<Vector<double>> &support_point_values,
+    std::vector<double> &              nodal_values) const
```

Comment:
This function is also independently correct and we can merge that in a separate PR if you'd like.